### PR TITLE
Add close strategy ixn

### PR DESCRIPTION
**Pull request by alittlezz**

### CODE REVIEW

**Comment by silviutroscot** [comment]
Can you do a test in `kamino.test.ts` so we see it works and it acts like an example for the UI?

**Review by silviutroscot — APPROVED** [review]
LGTM! I think we can wait until the new release is out so we can see the pipeline is green

**Review by y2kappa — APPROVED** [review]

File: packages/kamino-sdk/src/constants/pubkeys.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { PublicKey } from '@solana/web3.js';
 
+export const DEFAULT_PUBKEY = new PublicKey(0);
```
Comment (silviutroscot) [inline packages/kamino-sdk/src/constants/pubkeys.ts:3]: there is a `Pubkey.default`, you don't need this

File: packages/kamino-sdk/src/Kamino.ts [hunk 2]
```diff
@@ -2858,6 +2860,122 @@ export class Kamino {
...
+    if (!strategyState.position.equals(DEFAULT_PUBKEY)) {
+      oldPositionOrBaseVaultAuthority = strategyState.position;
+      oldPositionMintOrBaseVaultAuthority = strategyState.positionMint;
+      oldPositionTokenAccountOrBaseVaultAuthority = strategyState.positionTokenAccount;
+    }
+    let userTokenAAta = getAssociatedTokenAddress(strategyState.tokenAMint, strategyState.adminAuthority);
+    let userTokenBAta = getAssociatedTokenAddress(strategyState.tokenBMint, strategyState.adminAuthority);
+    let reward0Vault = strategyState.baseVaultAuthority;
+    let userReward0Ata = strategyState.baseVaultAuthority;
+    if (!strategyState.reward0Vault.equals(DEFAULT_PUBKEY) && strategyState.reward0Decimals.toNumber() > 0) {
```
Comment (silviutroscot) [inline packages/kamino-sdk/src/Kamino.ts]: can you make an util where you pass the vault and decimals and does this check or maybe even returns directly the ATA you need? it is very duplicated

File: packages/kamino-sdk/tests/kamino.test.ts [hunk 3]
```diff
@@ -1710,6 +1713,106 @@ describe('Kamino SDK Tests', () => {
...
+      fixtures.kaminoProgramId,
+      WHIRLPOOL_PROGRAM_ID,
+      LOCAL_RAYDIUM_PROGRAM_ID
+    );
+    // Create a new strategy
+    let txStrategyCreate = createTransactionWithExtraBudget(signer.publicKey);
+    const newOrcaStrategy = Keypair.generate();
+    const createStrategyAccountIx = await kamino.createStrategyAccount(signer.publicKey, newOrcaStrategy.publicKey);
+    txStrategyCreate.add(createStrategyAccountIx);
+    let orcaStrategyIx = await kamino.createStrategy(
```
[the remark this answers is not in the document]
Comment (alittlezz) [reply packages/kamino-sdk/tests/kamino.test.ts:1779]: at L1789

File: packages/kamino-sdk/src/constants/pubkeys.ts (hunk 1 above)
Comment (alittlezz) [reply packages/kamino-sdk/src/constants/pubkeys.ts]: done

File: packages/kamino-sdk/tests/kamino.test.ts [hunk 4]
```diff
@@ -1710,6 +1713,106 @@ describe('Kamino SDK Tests', () => {
...
+      const position = await kamino.getPositionRangeOrca(
+        strategyState.position,
+        strategyState.tokenAMintDecimals.toNumber(),
+        strategyState.tokenBMintDecimals.toNumber()
+      );
+      expect(position).to.be.null;
+    } catch (error) {
+      console.log(`Fetching closed position got err ${error}`);
+    }
+  });
```
[the remark this answers is not in the document]
Comment (alittlezz) [reply packages/kamino-sdk/tests/kamino.test.ts:1815]: no need to duplicate bpf tests, this should only be as a reference for FE

[the remark this answers is not in the document]
Comment (silviutroscot) [reply packages/kamino-sdk/tests/kamino.test.ts:1815]: i was thinking that we may want to do that in FE as well, for people who have a strat to `withdrawAndCloseStrat` with a single button

File: packages/kamino-sdk/src/Kamino.ts (hunk 2 above)
Comment (alittlezz) [reply packages/kamino-sdk/src/Kamino.ts]: refactored the condition in a separate func

File: packages/kamino-sdk/tests/kamino.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (silviutroscot) [reply packages/kamino-sdk/tests/kamino.test.ts:1779]: that just returns an instruction

**Review by silviutroscot — APPROVED** [review]

**Pull request merged**